### PR TITLE
[wip] Bump vishvananda/netlink to 1.0.0

### PR DESCRIPTION
**Pull request by thaJeztah**
Full diff: https://github.com/vishvananda/netlink/compare/b2de5d10e38ecce8607e6b438b6d174f389a004e...v1.0.0

Changes included:

- https://github.com/vishvananda/netlink/pull/283 Allow index specification at link creation time
- https://github.com/vishvananda/netlink/pull/286 replace syscall with golang.org/x/sys/unix
  - related: https://github.com/vishvananda/netlink/pull/330 Use IFF_MULTI_QUEUE from x/sys/unix to define TUNTAP_MULTI_QUEUE
  - related: https://github.com/vishvananda/netlink/pull/332 Use IFLA_* constants from x/sys/unix
- https://github.com/vishvananda/netlink/pull/287 Fix index out of range when no metadata for gretap
- https://github.com/vishvananda/netlink/pull/281 added encapsulation attributes for Iptun and Sittun to support SIT tunnels
- https://github.com/vishvananda/netlink/pull/288 Expose xfrm state's statistics
- https://github.com/vishvananda/netlink/pull/290 Support invert in ip rules
- https://github.com/vishvananda/netlink/pull/282 Support LWTUNNEL_ENCAP_SEG6 
- https://github.com/vishvananda/netlink/commit/63ca7e48f59f463578ae6b6e839bef6906f048e2 Support setting and retrieving route MTU/AdvMSS
- https://github.com/vishvananda/netlink/pull/259 Fix CalcRtable array parameter bug
- https://github.com/vishvananda/netlink/pull/285 added support for Foo-over-UDP netlink calls
- https://github.com/vishvananda/netlink/pull/225 Support num{tx,rx}queues and udp6zerocsum{tx,rx}
- https://github.com/vishvananda/netlink/pull/293 tuntap: Add multiqueue support
- https://github.com/vishvananda/netlink/pull/227 Retrieve VLAN ID when listing neighbour
- https://github.com/vishvananda/netlink/pull/301 Fix LinkAdd for sit tunnel on 3.10 kernel
- https://github.com/vishvananda/netlink/pull/303 Add support for managing source MACVLANs
- https://github.com/vishvananda/netlink/pull/307 Two functions: one for adding bond slave, one for getting veth peer index
- https://github.com/vishvananda/netlink/pull/308 Eliminate cgo from netlink
- https://github.com/vishvananda/netlink/pull/309 Don't overwrite the XDP file descriptor with flags
- https://github.com/vishvananda/netlink/pull/313 Fix reference to BPF instructions (on Kernel 4.13)
- https://github.com/vishvananda/netlink/pull/311 Add Matchall filter
- https://github.com/vishvananda/netlink/pull/315 Send IFA_CACHEINFO when setting up addresses
- https://github.com/vishvananda/netlink/pull/317 Support IPv6 GRE Tun and Tap
- https://github.com/vishvananda/netlink/pull/318 Add List option to RouteSubscribeWithOptions, AddrSubscribeWithOptions, and LinkSubscribeWithOptions
- https://github.com/vishvananda/netlink/pull/321 Add Fq and Fq_Codel Qdisc support

ping @aboch @fcrisciani PTAL

Looks like with this changes, only the `system/meminfo` code is needing CGO https://github.com/docker/libnetwork/blob/873af3416744d02ea044cf02a46d6dc3857b9062/vendor/github.com/docker/docker/pkg/system/meminfo_solaris.go#L1, which could probably be updated

### CODE REVIEW

**Comment by thaJeztah** [comment]
Looks like I need to update `golang/x/sys/` accordingly;

```
# github.com/docker/libnetwork/vendor/github.com/vishvananda/netlink
vendor/github.com/vishvananda/netlink/bridge_linux.go:22:27: undefined: unix.IFLA_EXT_MASK
vendor/github.com/vishvananda/netlink/bridge_linux.go:38:9: undefined: unix.IFLA_AF_SPEC
vendor/github.com/vishvananda/netlink/bridge_linux.go:90:21: undefined: unix.IFLA_AF_SPEC
vendor/github.com/vishvananda/netlink/link_linux.go:380:23: undefined: unix.IFLA_VFINFO_LIST
vendor/github.com/vishvananda/netlink/link_linux.go:410:23: undefined: unix.IFLA_VFINFO_LIST
vendor/github.com/vishvananda/netlink/link_linux.go:440:23: undefined: unix.IFLA_VFINFO_LIST
vendor/github.com/vishvananda/netlink/link_linux.go:471:23: undefined: unix.IFLA_VFINFO_LIST
vendor/github.com/vishvananda/netlink/link_linux.go:505:23: undefined: unix.IFLA_VFINFO_LIST
vendor/github.com/vishvananda/netlink/link_linux.go:632:23: undefined: unix.IFLA_NET_NS_FD
vendor/github.com/vishvananda/netlink/link_linux.go:1004:28: undefined: unix.IFLA_NUM_TX_QUEUES
vendor/github.com/vishvananda/netlink/link_linux.go:1004:28: too many errors
# github.com/docker/libnetwork/ipvs
ipvs/ipvs.go:94:32: cannot use &tv (type *syscall.Timeval) as type *unix.Timeval in argument to sock.SetSendTimeout
ipvs/ipvs.go:98:35: cannot use &tv (type *syscall.Timeval) as type *unix.Timeval in argument to sock.SetReceiveTimeout
make: *** [build-local] Error 2
Makefile:27: recipe for target 'build-local' failed
make: *** [circle-ci-build] Error 2
```

File: controller.go [hunk 1]
```diff
@@ -1255,7 +1256,7 @@ func (c *controller) loadDriver(networkType string) error {
 	}
 
 	if err != nil {
-		if err == plugins.ErrNotFound {
+		if errors.Cause(err) == plugins.ErrNotFound {
```
Comment (thaJeztah) [inline controller.go]: This fixes:

```
--- FAIL: TestUnknownDriver (15.00s)
	libnetwork_test.go:213: Did not fail with expected error. Actual error: could not find plugin unknowndriver in v1 plugin registry: plugin not found

--- FAIL: TestNilRemoteDriver (15.00s)
	libnetwork_test.go:225: Did not fail with expected error. Actual error: could not find plugin framerelay in v1 plugin registry: plugin not found
```

File: controller.go [hunk 2]
```diff
@@ -1274,7 +1275,7 @@ func (c *controller) loadIPAMDriver(name string) error {
...
```
Comment (thaJeztah) [inline controller.go]: Similar to the above, but not sure if there's a test case covering this;

```
--- FAIL: TestUnknownDriver (15.00s)
	libnetwork_test.go:213: Did not fail with expected error. Actual error: could not find plugin unknowndriver in v1 plugin registry: plugin not found

--- FAIL: TestNilRemoteDriver (15.00s)
	libnetwork_test.go:225: Did not fail with expected error. Actual error: could not find plugin framerelay in v1 plugin registry: plugin not found
```

**Comment by thaJeztah** [comment]
Noticed this warning; will have to look into this one;

```
=== RUN   TestValidRemoteDriver
time="2018-03-19T18:21:21Z" level=warning msg="driver error deleting network dummy : NetworkDriver.DeleteNetwork: 404 page not found\n"
```

This looks like an actual failure:

```
=== RUN   TestRequestReleaseAddressDuplicate
--- FAIL: TestRequestReleaseAddressDuplicate (0.00s)
	allocator_test.go:1436: IP 198.168.0.226/23 was previously allocated
```

File: controller.go (hunk 1 above)
Comment (thaJeztah) [reply controller.go]: Wonder if this is an actual issue currently; this was revealed due to docker/docker being bumped here, but likely is an issue in the current version in that repository

**Comment by fcrisciani** [comment]
cc @abhi looks like one of the IPAM tests failed here

**Comment by amoghe** [comment]
No <3 for this?

**Comment by roffe** [comment]
did this die in usersub?

**Comment by euanh** [comment]
Superseded by #2366

**Pull request closed without merging**